File: Editor/src/main/java/org/turbomedia/turboedit/editor/panes/timeline/UpperBar.java
package org.turbomedia.turboedit.editor.panes.timeline;

import javafx.geometry.Insets;
import javafx.scene.Cursor;
import javafx.scene.control.MenuButton;
import javafx.scene.control.MenuItem;
import javafx.scene.layout.*;
import javafx.scene.paint.Color;
import javafx.scene.text.Font;
import javafx.scene.text.FontWeight;
import javafx.scene.text.Text;
import javafx.scene.text.TextAlignment;
import org.kordamp.ikonli.fluentui.FluentUiFilledAL;
import org.kordamp.ikonli.fluentui.FluentUiFilledMZ;
import org.kordamp.ikonli.javafx.FontIcon;
import org.turbomedia.turboedit.editor.misc.StyleManager;

import static org.turbomedia.turboedit.editor.misc.Locale.GetText;

public class UpperBar extends HBox {

    public UpperBar() {
        var FONT_NAME = Font.getDefault().getName();
        var timestamp = new Text("00:00:00.0000");

        timestamp.setFont(Font.font(FONT_NAME, FontWeight.BLACK, 24));
        timestamp.setTextAlignment(TextAlignment.CENTER);

        var newButton = new MenuButton();
        newButton.setCursor(Cursor.HAND);
        newButton.setGraphic(FontIcon.of(FluentUiFilledAL.ADD_SQUARE_24));
        {
            var audioItem = new MenuItem(GetText("timeline.upper_bar.add.audio"));
            audioItem.setGraphic(FontIcon.of(FluentUiFilledMZ.SPEAKER_24));

            newButton.getItems().add(audioItem);
        }

        var settingsButton = new MenuButton();
        settingsButton.setCursor(Cursor.HAND);
        settingsButton.setGraphic(FontIcon.of(FluentUiFilledMZ.SETTINGS_24));
        {
        }

        setSpacing(8);
        setPadding(new Insets(8));
        setMaxWidth(Double.MAX_VALUE);
        HBox.setHgrow(this, Priority.ALWAYS);

        StyleManager.RegisterEvent((theme) -> {
            Color color = switch (theme) {
                case DARK -> Color.color(1, 1, 1, 0.05);
                case LIGHT -> Color.color(0, 0, 0, 0.05);
            };

            var background = new Background(new BackgroundFill(color, new CornerRadii(10), Insets.EMPTY));

            setBackground(background);
        });

        getChildren().add(timestamp);
        getChildren().add(newButton);
        getChildren().add(settingsButton);
    }
}

File: Editor/src/main/kotlin/turboedit/editor/panes/timeline/UpperBar.kt
package turboedit.editor.panes.timeline

import javafx.geometry.Insets
import javafx.scene.Cursor
import javafx.scene.control.MenuButton
import javafx.scene.control.MenuItem
import javafx.scene.layout.*
import javafx.scene.paint.Color
import javafx.scene.paint.Paint
import javafx.scene.text.Font
import javafx.scene.text.FontWeight
import javafx.scene.text.Text
import javafx.scene.text.TextAlignment
import org.kordamp.ikonli.fluentui.FluentUiFilledAL
import org.kordamp.ikonli.fluentui.FluentUiFilledMZ
import org.kordamp.ikonli.javafx.FontIcon
import turboedit.editor.events.EventSystem.EventAction
import turboedit.editor.events.EventSystem.registerListener
import turboedit.editor.events.EventType
import turboedit.editor.events.ThemeChangedEventData
import turboedit.editor.misc.Locale
import turboedit.editor.misc.StyleManager
import turboedit.editor.misc.StyleManager.callEvent

class UpperBar : HBox() {
    init {
        val fontName = Font.getDefault().name
        val timestamp = Text("00:00:00.0000")

        timestamp.font = Font.font(fontName, FontWeight.BLACK, 24.0)
        timestamp.textAlignment = TextAlignment.CENTER

        val newButton = MenuButton()
        newButton.cursor = Cursor.HAND
        newButton.graphic = FontIcon.of(FluentUiFilledAL.ADD_SQUARE_24)

        val audioItem = MenuItem(Locale.getText("timeline.upper_bar.add.audio"))
        audioItem.graphic = FontIcon.of(FluentUiFilledMZ.SPEAKER_24)
        newButton.items.add(audioItem)


        val settingsButton = MenuButton()
        settingsButton.cursor = Cursor.HAND
        settingsButton.graphic = FontIcon.of(FluentUiFilledMZ.SETTINGS_24)

        spacing = 8.0
        padding = Insets(8.0)
        maxWidth = Double.Companion.MAX_VALUE
        setHgrow(this, Priority.ALWAYS)

        registerListener(EventType.THEME_CHANGED, EventAction { dat: Any? ->
            val data = dat as ThemeChangedEventData?
            val color = when (data!!.theme) {
                StyleManager.Theme.DARK -> Color.color(1.0, 1.0, 1.0, 0.05)
                StyleManager.Theme.LIGHT -> Color.color(0.0, 0.0, 0.0, 0.05)
                else -> {}
            }

            val background = Background(BackgroundFill(color as Paint?, CornerRadii(10.0), Insets.EMPTY))
            setBackground(background)
        })
        callEvent()

        children.add(timestamp)
        children.add(newButton)
        children.add(settingsButton)
    }
}
